examples/shallow_water_Bickley_jet.jl

```diff
@@ -34,9 +34,9 @@ using Oceananigans.Models: ShallowWaterModel
 Lx, Ly, Lz = 2π, 20, 1
 Nx, Ny = 128, 128
 
-grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
-                            x = (0, Lx), y = (-Ly/2, Ly/2), z = (0, Lz),
-                            topology = (Periodic, Bounded, Bounded))
+grid = RegularRectilinearGrid(size = (Nx, Ny),
+                            x = (0, Lx), y = (-Ly/2, Ly/2),
+                            topology = (Periodic, Bounded, Flat))
```

Comment:
🥰

Reply:
Thanks @ali and agree that it should be integrated.  I'm happy to try `NonFlat` if we think it's going to be not too difficult, and that way we can make sure the other models are working fine with this new change.

Reply:
Will save `NonFlat` for a future PR